api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java

```diff
@@ -115,6 +115,16 @@
      */
     <T> ConfigBuilder withConverter(Class<T> type, int priority, Converter<T> converter);
 
+    /**
+     * Set the config profile.
+     * Once the config profile is set, the config property {@code %profile.config.name} overrides the property {@code config.name} 
+     * in the same config source. 
+     * The config value for the property {@code config.name} is retrieved from a config source with the highest ordinal. 
+     * The value is set by the property {@code %profile.config.name} if exists. Otherwise, the value is taken from the property {@code config.name}.
+     * @param profile the config profile
+     * @return
```

Comment:
```suggestion
     * Set the configuration profile.
     * Once the configuration profile is set, the configuration property {@code %profile.config.name} overrides the property {@code config.name} 
     * in the same configuration source. 
     * The configuration value for the property {@code config.name} is retrieved from a configuration source with the highest ordinal.
     * The value is set by the property {@code %profile.config.name} if exists. Otherwise, the value is taken from the property {@code config.name}.
     * @param profile the configuration profile name (must not be {@code null})
     * @return this builder
```

Also the resolution of #553 would affect the wording here.

Reply:
updated.

Reply:
When I tried to add tests for this spi, I suggest against the idea adding this method, because it will create more confusions. Setting `mp.config.profile` is better set via the config sources to avoid unnecessary conflicts of setting profile via a method or via a property.